pymc3/step_methods/hmc/nuts.py

```diff
@@ -352,24 +306,41 @@ def extend(self, direction):
 
         return diverging, turning
 
-    def _build_subtree(self, left, depth, epsilon):
-        if depth == 0:
+    def _single_step(self, left, epsilon):
+        """Perform a leapfrog step and handle error cases."""
+        try:
             right = self.leapfrog(left.q, left.p, left.q_grad, epsilon)
+        except linalg.LinalgError as error:
+            error_msg = "LinAlgError during leapfrog step."
+        except ValueError as error:
+            # Raised by many scipy.linalg functions
+            if error.args[0].lower() == 'array must not contain infs or nans':
```

Comment:
this is a little dangerous: it is brittle to changes in `scipy.linalg`, and will throw an `IndexError` if there is no message.  Can you check directly if the array contains infs or nans?